From 26e6d1a15e6f792c5a08942333d536a08918e4f9 Mon Sep 17 00:00:00 2001
From: Sergey Vinokurov <serg.foo@gmail.com>
Date: Tue, 21 Aug 2018 09:27:06 +0100
Subject: [PATCH] Fix tracking of freed global values when module assertions
 are enabled

---
 src/emacs-module.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/emacs-module.c b/src/emacs-module.c
index 1b19e8033d..23c24d4181 100644
--- a/src/emacs-module.c
+++ b/src/emacs-module.c
@@ -337,17 +337,17 @@ module_free_global_ref (emacs_env *env, emacs_value ref)
       for (Lisp_Object tail = global_env_private.values; CONSP (tail);
            tail = XCDR (tail))
         {
-          emacs_value global = XSAVE_POINTER (XCAR (globals), 0);
+          emacs_value global = XSAVE_POINTER (XCAR (tail), 0);
           if (global == ref)
             {
               if (NILP (prev))
                 global_env_private.values = XCDR (globals);
               else
-                XSETCDR (prev, XCDR (globals));
+                XSETCDR (prev, XCDR (tail));
               return;
             }
           ++count;
-          prev = globals;
+          prev = tail;
         }
       module_abort ("Global value was not found in list of %"pD"d globals",
                     count);
-- 
2.18.0

